File: examples/ts/langchain/agent.ts
import { readFileSync } from "node:fs";
import * as readline from "node:readline";
import { ChatAnthropic } from "@langchain/anthropic";
import { createReactAgent } from "@langchain/langgraph/prebuilt";
import { HumanMessage } from "@langchain/core/messages";
import { createVelaToolkit } from "vela-sdk/adapters/langchain";

const workflowYaml = readFileSync("./workflows/project-setup.yaml", "utf-8");

const { tools } = createVelaToolkit({
  workflows: [workflowYaml],
});

const llm = new ChatAnthropic({ model: "claude-sonnet-4-20250514" });
const agent = createReactAgent({ llm, tools });

// Interactive loop
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

console.log("Vela + LangChain Agent");
console.log('Type a message. Type "exit" to quit.\n');

function ask() {
  rl.question("You: ", async (input) => {
    if (input.toLowerCase() === "exit") {
      rl.close();
      return;
    }
    const result = await agent.invoke({
      messages: [new HumanMessage(input)],
    });
    const messages = result.messages;
    for (let i = messages.length - 1; i >= 0; i--) {
      if (messages[i]._getType() === "ai" && messages[i].content) {
        console.log(`Agent: ${messages[i].content}\n`);
        break;
      }
    }
    ask();
  });
}
ask();

File: examples/python/langchain/agent.py
"""LangChain ReAct agent with Vela workflow tools."""

import asyncio

from langchain_anthropic import ChatAnthropic
from langgraph.prebuilt import create_react_agent

from vela_sdk.langchain import VelaToolkit

# Load workflows from YAML and get LangChain-compatible tools
toolkit = VelaToolkit(workflows_dir="./workflows/")
tools = toolkit.get_tools()

# Create a ReAct agent with Claude and Vela tools
llm = ChatAnthropic(model="claude-sonnet-4-20250514")
agent = create_react_agent(llm, tools)


async def main():
    print("Vela + LangChain Agent")
    print('Type a message. Type "exit" to quit.\n')

    while True:
        user_input = input("You: ")
        if user_input.lower() in ("quit", "exit"):
            break

        result = await agent.ainvoke(
            {"messages": [{"role": "user", "content": user_input}]}
        )

        # Print the last AI message
        for msg in reversed(result["messages"]):
            if msg.type == "ai" and msg.content:
                print(f"Agent: {msg.content}\n")
                break


if __name__ == "__main__":
    asyncio.run(main())
